braid/_braid_status.c

```diff
@@ -352,7 +352,61 @@ braid_StatusSetRFactor(braid_Status status,
                        braid_Real   rfactor
                        )
 {
-   _braid_StatusElt(status, rfactor) = rfactor;
+   braid_Int  level = _braid_StatusElt(status, level);
+
+   /* Only set the rfactor on level 0 */
+   if (level == 0)
+   {
+      _braid_Grid      **grids    = _braid_StatusElt(status, grids);
+      braid_Int         *rfactors = _braid_StatusElt(status, rfactors);
+      braid_Int          index    = _braid_StatusElt(status, idx);
+      braid_Int          ilower   = _braid_GridElt(grids[level], ilower);
+      braid_Int          ii       = index+1 - ilower;
+
+      /* Set refinement factor */
+      rfactors[ii] = rfactor;
+   }
+
+   return _braid_error_flag;
+}
+
+braid_Int
+braid_StatusSetRefinementDtValues(braid_Status  status,
+                                  braid_Real    rfactor,
+                                  braid_Real   *rdtarray
+                                  )
+{
+   braid_Int  level = _braid_StatusElt(status, level);
+
+   /* Only set the rfactor on level 0 */
+   if (level == 0)
+   {
+      _braid_Grid      **grids    = _braid_StatusElt(status, grids);
+      braid_Int         *rfactors = _braid_StatusElt(status, rfactors);
+      braid_Int          index    = _braid_StatusElt(status, idx);
+      braid_Int          ilower   = _braid_GridElt(grids[level], ilower);
+      braid_Int          ii       = index+1 - ilower;
+
+      /* Set refinement factor */
+      rfactors[ii] = rfactor;
+
+      /* Store dt values */
+      if (rfactor > 1)
+      {
+         braid_Real  **rdtvalues = _braid_StatusElt(status, rdtvalues);
+         braid_Int     j;
+
+         if (rdtvalues[ii] == NULL)
```

Comment:
Maybe we want to first test if rdtvalues[ii] != NULL, and if so, free the space. This would ensure that rdtvalues[ii] always has the correct length (rfactor-1) for the copy that comes afterwards. We could possibly create a segfault otherwise, e.g. if rdtvalues[ii] has been allocated elsewhere using a different rfactor. Not sure if that would ever happen, but is still somewhat unsafe.

Reply:
Good point!  Fixed.